### PR TITLE
Completed object-manipulation-3.

### PR DESCRIPTION
**Pull request by alexdenbaugh**
object-manipulation-3-1
object-manipulation-3-2

### CODE REVIEW

File: object-manipulation-3/main.js [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+    rank: 2,
+    suit: 'hearts'
+  }
+];
+
+function playCards(players, cardDeck, cardsPerHand) {
+  var shuffledDeck = [];
+  var randomNumber;
+  var minNumber = Math.ceil(0);
+  var maxNumber = Math.floor(cardDeck.length);
```
Comment (codingcodymiller) [inline object-manipulation-3/main.js:236]: Why the `ceil` and `floor` here?

File: object-manipulation-3/main.js [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+  var j;
+  var cardCount = 0;
+  var playerScore = 0;
+  var winner = { name: '', score: 0 };
+  var tieBreaker = [];
+  var whoTied = [];
+  while (cardDeck.length > shuffledDeck.length) {
+    randomNumber = Math.floor(Math.random() * (maxNumber - minNumber) + minNumber);
+    if (!shuffledDeck.includes(cardDeck[randomNumber])) {
```
Comment (codingcodymiller) [inline object-manipulation-3/main.js:245]: So does this loop just keep going until the loop manages to randomly get every card in the deck? That could take a long time. I would try to look into some ways you could just randomize the array. Make sure you understand how the code works, though!

File: object-manipulation-3/main.js [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+  }
+  for (var i = 0; i < players.length; i++) {
+    players[i].hand = [];
+  }
+  for (i = 0; i < cardsPerHand; i++) {
+    for (j = 0; j < players.length; j++) {
+      players[j].hand.push(shuffledDeck[cardCount]);
+      cardCount++;
+    }
+  }
```
Comment (codingcodymiller) [inline object-manipulation-3/main.js:257]: Is this to give each player their initial hand? You might be able to do this without the need for less moving pieces. You can just remove cards from the `shuffledDeck` as they get dealt, so you can always "deal" the same index.

File: object-manipulation-3/main.js [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+  for (i = 0; i < players.length; i++) {
+    for (j = 0; j < cardsPerHand; j++) {
+      if (players[i].hand[j].rank === 'ace') {
+        playerScore = playerScore + 13;
+      } else if (players[i].hand[j].rank === 'king' || players[i].hand[j].rank === 'queen' || players[i].hand[j].rank === 'jack') {
+        playerScore = playerScore + 10;
+      } else {
+        playerScore = playerScore + players[i].hand[j].rank;
+      }
```
Comment (codingcodymiller) [inline object-manipulation-3/main.js:266]: What about using an object that has a key for each card type, and a value of its card worth? That should simplify this by quite a bit.

**Review by codingcodymiller — CHANGES_REQUESTED** [review]
Great start here, dude!! Please take a look at my feedback for some areas to address

**Review by scbowler — APPROVED** [review]
Great job implementing Cody's feedback, looks good! 👍 

When time permits I would recommend trying to re do this in OOP as an additional challenge.

**Pull request merged**